packages/common/src/types/semanticLayer.ts

```diff
@@ -149,9 +149,80 @@ export type SemanticLayerQueryPayload = {
     context: 'semanticViewer';
 };
 
-export enum SemanticLayerStringFilterOperator {
+// Helper functions and constants
+const SEMANTIC_LAYER_DEFAULT_QUERY_LIMIT = 500;
+
+export const isSemanticLayerTimeDimension = (
+    field: SemanticLayerField,
+): field is SemanticLayerTimeDimension => 'granularity' in field;
+
+export function getDefaultedLimit(
+    maxQueryLimit: number,
+    queryLimit?: number,
+): number {
+    return Math.min(
+        queryLimit ?? SEMANTIC_LAYER_DEFAULT_QUERY_LIMIT,
+        maxQueryLimit,
+    );
+}
+
+// Semantic Layer Scheduler Job
+
+export const semanticLayerQueryJob = 'semanticLayer';
+
+export type SemanticLayerJobStatusSuccessDetails = {
+    fileUrl: string;
+    columns: string[];
+};
+
+export type SemanticLayerJobStatusErrorDetails = {
+    error: string;
+    charNumber?: number;
+    lineNumber?: number;
+    createdByUserUuid: string;
+};
+
+export type ApiSemanticLayerJobStatusResponse = {
+    status: 'ok';
+    results: {
+        status: SchedulerJobStatus;
+        details:
+            | SemanticLayerJobStatusSuccessDetails
+            | SemanticLayerJobStatusErrorDetails;
+    };
+};
+
+export type ApiSemanticLayerJobSuccessResponse =
+    ApiSemanticLayerJobStatusResponse & {
+        results: {
+            status: SchedulerJobStatus.COMPLETED;
+            details: SemanticLayerJobStatusSuccessDetails;
+        };
+    };
+
+export function isSemanticLayerJobErrorDetails(
+    results?: ApiSemanticLayerJobStatusResponse['results']['details'],
+): results is SemanticLayerJobStatusErrorDetails {
+    return (results as SemanticLayerJobStatusErrorDetails).error !== undefined;
+}
+
+export const isApiSemanticLayerJobSuccessResponse = (
+    response: ApiSemanticLayerJobStatusResponse['results'] | ApiError,
+): response is ApiSemanticLayerJobSuccessResponse['results'] =>
+    response.status === SchedulerJobStatus.COMPLETED;
```

Comment:
if you follow the previous comment - you don't need to have this at all since it will "typenarrow" based on the `status`